### PR TITLE
Update some modules to JUnit 5 [ECR-642]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview
Update all modules except `ejb-core` and `proofs` to JUnit 5.

---
See: https://jira.bf.local/browse/ECR-642 

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] The [continuous integration build](https://www.travis-ci.org/exonum/exonum-java-binding) passes

### CODE REVIEW

File: exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java [hunk 1]
```diff
@@ -128,27 +121,29 @@ public void submitValidTransaction(TestContext context) {
     post(ApiController.SUBMIT_TRANSACTION_PATH)
         .sendJsonObject(
             new JsonObject(messageJson),
-            context.asyncAssertSuccess(
-                r -> {
+            context.succeeding(
+                r -> context.verify(() -> {
```
Comment (bullet-tooth) [inline exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java]: `r` to `result`

File: exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java [hunk 2]
```diff
@@ -128,27 +121,29 @@ public void submitValidTransaction(TestContext context) {
...
                   // Check the response status
                   int statusCode = r.statusCode();
-                  context.assertEquals(HTTP_OK, statusCode);
+                  assertEquals(HTTP_OK, statusCode);
```
Comment (bullet-tooth) [inline exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java]: `assertThat`

File: exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java [hunk 3]
```diff
@@ -164,56 +159,62 @@ public void submitTransactionWhenInternalServerErrorIsThrown(TestContext context
     post(ApiController.SUBMIT_TRANSACTION_PATH)
         .sendJsonObject(
             new JsonObject(messageJson),
-            context.asyncAssertSuccess(ar -> {
-              context.verify(v -> {
-                assertThat(ar.statusCode()).isEqualTo(HTTP_INTERNAL_ERROR);
-                verify(service).submitTransaction(transaction);
-              });
-            }));
+            context.succeeding(ar -> context.verify(() -> {
```
Comment (bullet-tooth) [inline exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java]: `ar` is confusing

File: exonum-java-binding-fakes/src/test/java/com/exonum/binding/fakes/services/transactions/SetEntryTransactionIntegrationTest.java [hunk 4]
```diff
@@ -29,16 +29,21 @@
...
 import com.exonum.binding.storage.serialization.StandardSerializers;
 import com.exonum.binding.util.LibraryLoader;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
 
-public class SetEntryTransactionIntegrationTest {
+@DisabledOnOs(OS.LINUX)
```
Comment (bullet-tooth) [inline exonum-java-binding-fakes/src/test/java/com/exonum/binding/fakes/services/transactions/SetEntryTransactionIntegrationTest.java]: `@RequiresNativeLibrary` ?

File: exonum-java-binding-qa-service/src/test/java/com/exonum/binding/qaservice/ApiControllerIntegrationTest.java [hunk 5]
```diff
@@ -40,89 +41,75 @@
...
-        .listen(0, event -> {
-          assert event.succeeded();
+        .listen(0, context.succeeding(r -> {
 
           // Set the actual server port.
-          port = event.result().actualPort();
-          // Notify that the HTTP Server is accepting connections.
-          async.complete();
-        });
+          port = r.actualPort();
```
Comment (bullet-tooth) [inline exonum-java-binding-qa-service/src/test/java/com/exonum/binding/qaservice/ApiControllerIntegrationTest.java]: `r`

File: exonum-java-binding-qa-service/src/test/java/com/exonum/binding/qaservice/ApiControllerIntegrationTest.java [hunk 6]
```diff
@@ -191,15 +183,16 @@ public void submitCreateCounter_InternalServerError(TestContext context) {
         .thenThrow(error);
 
     post(ApiController.SUBMIT_CREATE_COUNTER_TX_PATH)
-        .sendForm(params, context.asyncAssertSuccess(ar -> {
-          context.verify(v -> {
+        .sendForm(params, context.succeeding(ar -> {
```
Comment (bullet-tooth) [inline exonum-java-binding-qa-service/src/test/java/com/exonum/binding/qaservice/ApiControllerIntegrationTest.java]: `ar` to something meaningful

File: exonum-java-binding-qa-service/src/test/java/com/exonum/binding/qaservice/ApiControllerIntegrationTest.java [hunk 7]
```diff
@@ -293,45 +286,51 @@ public void getCounter(TestContext context) {
...
+  void getCounter_InvalidIdFormat(VertxTestContext context) {
     String hash = "Invalid hexadecimal hash";
     String getCounterUri = getCounterUri(hash);
 
     get(getCounterUri)
-        .send(context.asyncAssertSuccess(ar -> context.verify(v -> {
-          assertThat(ar.statusCode()).isEqualTo(HTTP_BAD_REQUEST);
-          assertThat(ar.bodyAsString())
-              .startsWith("Failed to convert parameter (counterId):");
+        .send(context.succeeding(ar -> context.verify(() -> {
```
Comment (bullet-tooth) [inline exonum-java-binding-qa-service/src/test/java/com/exonum/binding/qaservice/ApiControllerIntegrationTest.java]: `ar`

File: exonum-java-binding-fakes/src/test/java/com/exonum/binding/fakes/services/transactions/SetEntryTransactionIntegrationTest.java (hunk 4 above)
Comment (dmitry-timofeev) [reply exonum-java-binding-fakes/src/test/java/com/exonum/binding/fakes/services/transactions/SetEntryTransactionIntegrationTest.java]: It is not in this PR.

File: exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java (hunk 2 above)
Comment (dmitry-timofeev) [reply exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java]: I'd keep as is at the moment — it's not the aim of this PR to migrate all usages of `assertEquals` to anything else.

File: exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java (hunk 3 above)
Comment (dmitry-timofeev) [reply exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java]: [async](Result|Response) — migrated to result/response where they are actually used.

**Comment by bullet-tooth** [comment]
well done

**Pull request merged**